snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -45,6 +45,7 @@ public class SnowflakeCatalog extends BaseMetastoreCatalog
     implements Closeable, SupportsNamespaces, Configurable<Object> {
   private static final String DEFAULT_CATALOG_NAME = "snowflake_catalog";
   private static final String DEFAULT_FILE_IO_IMPL = "org.apache.iceberg.io.ResolvingFileIO";
+  private static final String APP_IDENTIFIER = "iceberg-SDK";
```

Comment:
This looks like it's unrelated to the special characters.  Can we pull these change out?  I'm ok with the rest of the PR, but I'm not sure we want to use this particular identifier and it's easier to have a discussion about that separately.

Reply:
Removed the identifier.

Reply:
Removed the application identifier, will create a separate PR for it.